examples/dlint/js.rs

```diff
@@ -149,7 +149,7 @@ impl Plugin for JsRuleRunner {
   fn run(
     &mut self,
     context: &mut Context,
-    program: Program,
+    program: ProgramRef,
```

Comment:
The plugins now take a `ProgramRef` because that's what might be provided to `parse_with_ast`. This also eliminates cloning of a whole AST potentially a bunch of times.